File: backend/src/main/java/com/back/domain/adoption/dto/response/AdoptionResponseDto.java
package com.back.domain.adoption.dto.response;

import com.back.domain.adoption.entity.Adoption;
import com.back.domain.applicant.dto.response.ApplicantResponseDto;
import java.time.LocalDateTime;
import lombok.Builder;

@Builder
public record AdoptionResponseDto(
        Long adoptionId,
        Long petId,
        Long memberId,
        ApplicantResponseDto applicantInfo,
        String anotherPets,
        String experience,
        String title,
        String message,
        LocalDateTime createdAt

) {

    public static AdoptionResponseDto from(Adoption adoption) {
        return AdoptionResponseDto.builder()
                .adoptionId(adoption.getId())
                .petId(adoption.getPet().getId())
                .memberId(adoption.getMember().getId())
                .applicantInfo(ApplicantResponseDto.builder()
                        .id(adoption.getApplicant().getId())
                        .name(adoption.getApplicant().getName())
                        .phone(adoption.getApplicant().getPhone())
                        .email(adoption.getApplicant().getEmail())
                        .address(adoption.getApplicant().getAddress())
                        .build())
                .title(adoption.getTitle())
                .anotherPets(adoption.getAnotherPets())
                .experience(adoption.getExperience())
                .message(adoption.getMessage())
                .createdAt(adoption.getCreatedAt())
                .build();
    }
}

File: backend/src/main/java/com/back/domain/adoption/dto/response/AdoptionResponseDto.kt
package com.back.domain.adoption.dto.response

import com.back.domain.adoption.entity.Adoption
import com.back.domain.applicant.dto.response.ApplicantResponseDto
import java.time.LocalDateTime

data class AdoptionResponseDto(
    val adoptionId: Long?,
    val petId: Long?,
    val memberId: Long?,
    val applicantInfo: ApplicantResponseDto,
    val anotherPets: String?,
    val experience: String?,
    val title: String,
    val message: String?,
    val createdAt: LocalDateTime?
) {
    companion object {
        fun from(adoption: Adoption): AdoptionResponseDto? {
            return AdoptionResponseDto(
                adoptionId = adoption.id,
                petId = adoption.pet.id,
                memberId = adoption.member.id,
                applicantInfo = ApplicantResponseDto(
                    id = adoption.applicant.id,
                    name = adoption.applicant.name,
                    phone = adoption.applicant.phone,
                    email = adoption.applicant.email,
                    address = adoption.applicant.address
                ),
                title = adoption.title,
                anotherPets = adoption.anotherPets,
                experience = adoption.experience,
                message = adoption.message,
                createdAt = adoption.createdAt
            )
        }
    }
}
